### PR TITLE
Add tests and code quality checks

### PR DESCRIPTION
**Pull request by ikretz**
This PR adds `pytest` tests, `mypy` type checking, and `flake8` code quality checks to the repository.

These safety and code quality checks are enforced via the addition of pre-commit hooks.

### CODE REVIEW

File: tests/commands/test_npm.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+def test_npm_help_short():
+    """
+    Test that nothing is installed when the short form help option is present.
+    """
+    command_line = ["npm", "-h", "install", TEST_TARGET]
+    subprocess.run(command_line, check=True)
+    assert npm_list() == INIT_NPM_STATE
+
+
+def test_npm_help_long():
```
Comment (sobregosodd) [inline tests/commands/test_npm.py]: you can parametrize short and long to reduce code

File: tests/commands/test_npm.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    Test that nothing is installed when the long form help option is present.
+    """
+    command_line = ["npm", "--help", "install", TEST_TARGET]
+    subprocess.run(command_line, check=True)
+    assert npm_list() == INIT_NPM_STATE
+
+
+def test_npm_dry_run():
```
Comment (sobregosodd) [inline tests/commands/test_npm.py]: same here

File: tests/commands/test_npm.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    assert npm_list() == INIT_NPM_STATE
+
+    subprocess.run(command_line_2, check=True)
+    assert npm_list() == INIT_NPM_STATE
+
+    subprocess.run(command_line_3, check=True)
+    assert npm_list() == INIT_NPM_STATE
+
+
+def test_npm_dry_run_multiple():
```
Comment (sobregosodd) [inline tests/commands/test_npm.py]: same here, you can provide the command line directly and assert the same

File: tests/commands/test_npm_command.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+from scfw.commands.npm_command import NpmCommand
+
+from .test_npm import INIT_NPM_STATE, TEST_TARGET, npm_list
+
+
+def test_npm_command_would_install_basic_usage():
```
Comment (sobregosodd) [inline tests/commands/test_npm_command.py]: all the tests that do exactly the same but only its command line varies, you can parametrize

File: tests/commands/test_npm_command.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
+from scfw.commands.npm_command import NpmCommand
```
Comment (sobregosodd) [inline tests/commands/test_npm_command.py:3]: We are missing: 
* a test for a not existing package
* a known test for dependencies, for instance `requests@0.3.0`
    * wants exactly `axo eventemitter3 extendible hang loads node-http-xhr xhr-send`

File: tests/commands/test_pip.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
+import json
```
Comment (sobregosodd) [inline tests/commands/test_pip.py:1]: We are missing: 
* a test for a not existing package
* a known test for dependencies, for instance `requests@2.32.3`
    * wants exactly `urllib3 ...`

File: tests/verifiers/test_osv_verifier.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    ("sparxy", "1.0.3"),
+    ("dogbong", "1.0.1"),
+    ("@shennong/web-logger", "25.0.1"),
+    ("ajax-libary", "2.0.3"),
+    ("updated-tricks-v-bucks-generator-free_2023-pdz09", "4.2.7"),
+    ("cst-web-chat", "3.3.7"),
+]
+
+
+def test_osv_verifier_malicious_pip():
```
Comment (sobregosodd) [inline tests/verifiers/test_osv_verifier.py]: you can parametrize these both

Comment (ikretz) [reply tests/verifiers/test_osv_verifier.py]: How do you mean?

Comment (sobregosodd) [reply tests/verifiers/test_osv_verifier.py]:
```
@pytest.mark.parametrize("eco", [NPM, PIP])
def test_osv_verifier_malicious(eco:ecosystem):
```

Comment (ikretz) [reply tests/verifiers/test_osv_verifier.py]: Ohh, nice!  Gotcha, will do.

**Pull request merged**